velox/functions/sparksql/tests/RegexFunctionsTest.cpp

```diff
@@ -255,6 +260,18 @@ TEST_F(RegexFunctionsTest, regexpReplaceSimple) {
   auto result = testRegexpReplace("Hello World", "l", "L");
   EXPECT_EQ(result, output);
 }
+TEST_F(RegexFunctionsTest, badUTF8) {
+  std::string badUTF = "\xF0\x82\x82\xAC";
+  std::string badHalf = "\xF0\x82";
+  VELOX_ASSERT_THROW(
+      testingRegexpReplaceRows({badUTF}, {badHalf}, {"Bad"}), "invalid UTF-8");
+  // python converts above values to below and completes regexp_replace
```

Comment:
I'm not sure I understand this comment. Why is it taking about Python?

Reply:
This is an oversight on my end. Originally, this test case was a sanity checking test case to model behavior I observed in PySpark/python where I incorrectly noticed that PySpark _seemed_ to replace some bad UTF-8 values with parsable ones. Shown below

<img width="357" alt="image" src="https://github.com/facebookincubator/velox/assets/14911410/7a05e6a4-f771-460c-b084-47f4c68932ae">
<img width="155" alt="image" src="https://github.com/facebookincubator/velox/assets/14911410/957cfbec-8150-415a-8db7-0bb159e4c623">

As you can see in the byte output, it does not replace them but rather it just does not display \x82. 

I will change this test case to just make sure we throw an error when we encounter bad UTF-8 characters. RE2 simply does not allow these patterns to be compiled so a fallback to vanilla spark is needed as far as I can tell.